src/bytes_mut.rs

```diff
@@ -292,6 +275,12 @@ impl BytesMut {
             at,
             self.capacity(),
         );
+
+        if self.data.is_null() {
+            debug_assert_eq!(at, 0, "Empty Bytes is only splittable at the front");
```

Comment:
Is this possible to ever violate? since when `data.is_null()` then `capacity = 0` and the above assertion will check that?

Reply:
It shouldn't be possible to violate it. It's more just a guard and indicator in case of something was missed somewhere, or future changes are added. That's also why it is a `debug_assert` and not an `assert`

Reply:
Makes sense